bindings/ros/aditof_roscpp/src/message_factory.cpp

```diff
@@ -0,0 +1,20 @@
+#include "message_factory.h"
+
+aditof::Status MessageFactory::create(const std::shared_ptr<aditof::Camera> &camera, MessageType type, GenericMsg& msg)
+{
+    switch(type)
+    {
+        case sensor_msgs_PointCloud2:
+            msg.createMessage(camera);
+            return aditof::Status::OK;
+        case sensor_msgs_Image:
```

Comment:
if sensor_msgs_Image and others are not yet supported, don't include them here and in the enum. Otherwise clients could try to use them and discover at run-time that they are not supported. Just comment them in the enum and mark them as to-do.